src/megaapi_impl.cpp

```diff
@@ -18425,15 +18436,22 @@ void MegaApiImpl::updateBackups()
     }
 }
 
-unsigned MegaApiImpl::sendPendingTransfers()
+unsigned MegaApiImpl::sendPendingTransfers(TransferQueue *queue)
 {
     auto t0 = std::chrono::steady_clock::now();
     unsigned count = 0;
 
     SdkMutexGuard guard(sdkMutex);
     DBTableTransactionCommitter committer(client->tctable);
 
-    while(MegaTransferPrivate *transfer = transferQueue.pop())
+    TransferQueue &auxQueue = queue
+            ? *queue            // custom transferQueue used for folder uploads/downloads
+            : transferQueue;    // transfer queue of class MegaApiImpl
+
+    // if we are processing a custom queue, we need to process in one shot
+    bool timeout = !queue;
```

Comment:
It is not a timeout only, but also the amount of transfers processed. Better to name it differently. Maybe `canSplit` or alike